js_modules/dagit/packages/core/src/assets/usePartitionHealthData.tsx

```diff
@@ -469,14 +467,47 @@ export function usePartitionHealthData(assetKeys: AssetKey[], assetLastMateriali
       ]);
     };
     run();
-  }, [client, missingKeyJSON, assetLastMaterializedAt]);
+  }, [client, missingKey, assetLastMaterializedAt]);
```

Comment:
I'm not sure this is memoizable because it's an object containing a path and not a string, maybe let's leave this as-is for now and make a second PR with the JSON => no-JSON change when we have more time to test that the memoization is still working properly.

I think there are a few places in the UI where the asset keys are actually built into this array so they might be different objects between renders?

Reply:
Ah I see, sure. I made these changes when trying to adapt this hook but ended up abandoning it because having some stale health data and some not stale causes `mergeAssetHealth` to raise an exception.